index.js

```diff
@@ -1,23 +1,84 @@
 var util = require('util'),
     winston = require('winston'),
-    cloudwatchIntegration = require('./lib/cloudwatch-integration');
+    AWS = require('aws-sdk'),
+    _ = require('lodash');
 
 var CloudWatch = winston.transports.CloudWatch = function(options) {
-  this.name = 'CloudWatch';
+  this.name = options.name || 'CloudWatch';
   this.level = options.level || 'info';
+  this.logEvents = [];
+
+  this.options = options;
+
+  if (options.awsProfile && options.awsRegion) {
+    AWS.config.credentials = new AWS.SharedIniFileCredentials({profile: options.awsProfile});
+    this.cloudwatchlogs = new AWS.CloudWatchLogs({region: options.awsRegion});
+  } else if (awsAccessKeyId && awsSecretKey && options.awsRegion) {
```

Comment:
Ok no problems with that, once this is done I will merge, fix the tests and then release a new version on npm.